core/shared/src/main/scala/zio/macros/access/Accessible.scala

```diff
@@ -15,42 +15,70 @@
  */
 package zio.macros.access
 
+import com.github.ghik.silencer.silent
 import zio.macros.core.ModulePattern
 
 import scala.annotation.{ StaticAnnotation, compileTimeOnly }
 import scala.language.experimental.macros
 import scala.reflect.macros.whitebox.Context
 
 @compileTimeOnly("enable macro paradise to expand macro annotations")
-class accessible() extends StaticAnnotation {
+@silent("parameter value style in class accessible is never used")
+class accessible(style: String = "default") extends StaticAnnotation {
   def macroTransform(annottees: Any*): Any = macro AccessibleMacro.apply
 }
 
 private[access] class AccessibleMacro(val c: Context) extends ModulePattern {
   import c.universe._
 
+  sealed trait Style
+
+  object Style {
+    case object Default   extends Style
+    case object AliasOnly extends Style
+    case object TraitOnly extends Style
+  }
+
   def apply(annottees: c.Tree*): c.Tree = {
-    val trees       = extractTrees(annottees)
-    val module      = extractModule(trees.module)
-    val companion   = extractCompanion(trees.companion)
-    val service     = extractService(companion.body)
-    val capabilites = extractCapabilities(service)
-    val accessors   = generateCapabilityAccessors(trees.module, module.serviceName, capabilites)
-    val updated     = generateUpdatedCompanion(module, companion, accessors)
+
+    @silent("pattern var [^\\s]+ in method unapply is never used")
+    val style: Style = c.prefix.tree match {
+      case Apply(_, args) =>
+        val arg: String = args.collectFirst {
+          case q"$cfg" =>
+            c.eval(c.Expr[String](cfg))
+        }.getOrElse("default")
+
+        arg match {
+          case ">"       => Style.AliasOnly
```

Comment:
do we want to make the symbol configurable?

Reply:
I was thinking about it, but wasn't sure if there is such need. Now that you mentioned it I guess it does not hurt to expand the PR :)